homeassistant/components/leviosa_shades/cover.py

```diff
@@ -0,0 +1,180 @@
+"""The Leviosa Shades Zone base entity."""
+import logging
+
+from leviosapy import LeviosaShadeGroup as tShadeGroup, LeviosaZoneHub as tZoneHub
+import voluptuous as vol
+
+from homeassistant.components import cover
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv, entity_platform
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import (
+    BLIND_GROUPS,
+    DOMAIN,
+    MANUFACTURER,
+    MODEL,
+    SERVICE_NEXT_DOWN_POS,
+    SERVICE_NEXT_UP_POS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+# Estimated time it takes to complete a transition
+# from one state to another
+TRANSITION_COMPLETE_DURATION = 30
+PARALLEL_UPDATES = 1
+
+COVER_NEXT_POS_SCHEMA = vol.Schema({vol.Optional(ATTR_ENTITY_ID): cv.entity_ids})
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Leviosa shade groups."""
+    _LOGGER.debug(
+        "Setting up %s[%s]: %s",
+        entry.domain,
+        entry.title,
+        entry.entry_id,
+    )
+    hub_name = entry.title
+    hub_mac = entry.data["device_mac"]
+    hub_ip = entry.data["host"]
+    blind_groups = entry.data[BLIND_GROUPS]
+    _LOGGER.debug("Groups to create: %s", blind_groups)
+    hub = tZoneHub(
+        hub_ip=hub_ip, hub_name=hub_name, websession=async_get_clientsession(hass)
+    )
+    await hub.getHubInfo()  # Check all is good
```

Comment:
The implementation of `getHubInfo` is as follows（这部分代码在`leviosapy`）:
```python
async def getHubInfo(self):
        _LOGGER.debug("Getting HUB info from: %s", self.hub_ip)
        hub_resp = await self.get("")  # Query the root doc of the Hub to get Hub info
        if hub_resp:
            self.hub_fw_v = hub_resp["firmware"]
        else:
            self.hub_fw_v = "invalid"
```

You can see that this method just tries to update the `hub_fw_v` attribute, i.e., in order to go and update the version number of the hub, so what happens when this attribute is `invalid`?
Let's go back to `leviosa_shades`.

```python
async def validate_zone(hass: core.HomeAssistant, hub_address):
    """Ensure the Leviosa Zone is up and running and get the FW version."""
    try:
        _LOGGER.debug("Contacting Zone: %s", hub_address)
        hub = LeviosaZoneHub(
            hub_ip=hub_address,
            hub_name="tempZone",
            websession=async_get_clientsession(hass),
        )
        await hub.getHubInfo()
        _LOGGER.debug("Zone firmware v: %s", hub.fwVer)
    except HUB_EXCEPTIONS as err:
        raise CannotConnect from err
    if hub.fwVer == "invalid":
        raise CannotConnect
    return hub.fwVer
```

You can see that if `hub.fwVer` is `invalid`, then it triggers `CannotConnect` error.
Above, is the explanation to your query.

Reply:
I understand that the `CannotConnect` error can be raised in the config flow, but the `async_setup_entry` in `cover.py` is run every time Home Assistant starts I think so we need to handle the issue here too. 
`getHubInfo` should raise an error instead of using a side effect. There is also the possibility that `self.get` raises an error but it is not handled.

Reply:
I've updated the getHubInfo method to handle various exceptions inside, so it will be safer to use externally